lib/AST/NameLookup.cpp

```diff
@@ -1456,8 +1456,12 @@ DirectLookupRequest::evaluate(Evaluator &evaluator,
       auto allFound = evaluateOrDefault(
           ctx.evaluator, CXXNamespaceMemberLookup({cast<EnumDecl>(decl), name}),
           {});
-      for (auto found : allFound)
-        Table.addMember(found);
+      // Bypass the member lookup table if we find something in
+      // the original C++ namespace. We don't want to store
+      // the C++ decl in the lookup table as the decl can be referenced
+      // from multiple namespace declarations due to inline namespaces.
+      if (!allFound.empty())
+        return allFound;
```

Comment:
I'm surprised this works at all.. we need to combine `allFound` with whatever we get from `Table.find()` below.

Reply:
what kind of lookup would that scenario cover?

Reply:
maybe an ambiguous lookup between extension in Swift and the original C++ namespace? I can try that

Reply:
Sorry, I did kind of a rushed review. I should have explained more carefully. Here's the senerio I'm thinking of: 

```C++
namespace A {
inline namespace B {
void test(int, int);
}
}
```
```swift
extension A {
  func test(_ _: CInt) { }
}

func test() { A.test(0) } // not found
```

Reply:
Actually, I realized, you don't even need an inline namespace, I think. You could just get this to break by doing the following:
```
namespace A {
void test(int, int);
}

extension A {
  func test(_ _: CInt) { }
}

func test() { A.test(0) } // not found
```